src/test/groovy/graphql/schema/PropertyDataFetcherClassLoadingTest.groovy

```diff
@@ -0,0 +1,62 @@
+package graphql.schema
+
+import graphql.GraphQLException
+import graphql.Scalars
+import spock.lang.Specification
+
+class PropertyDataFetcherClassLoadingTest extends Specification {
+
+    GraphQLFieldDefinition fld(String fldName) {
+        return GraphQLFieldDefinition.newFieldDefinition().name(fldName).type(Scalars.GraphQLString).build()
+    }
+
+    static class BrokenClass {
+        static {
+            // this should prevent it from existing
+            throw new RuntimeException("No soup for you!")
+        }
+    }
+
+
+    static class TargetClass {
+
+        String getOkThings() {
+            return "ok"
+        }
+
+        BrokenClass getBrokenThings() {
+            return BrokenClass.cast(null)
+        }
+    }
```

Comment:
This class will cause a LinkageError however it will also fail on the reflection because `getBrokenThings` is fundamentally broken.

However what we see in the bug report is that the Metadatafactory fails but reflection does not.  Weirds me out.